### PR TITLE
Move util.lua to the specs directory

### PR DESCRIPTION
**Pull request by hugomg**
Now that this repo no longer has a Lua module, util.lua is the last Lua file remaining in the src directory. Let's move it to the spec directory, and leave only C stuff in the src.

### CODE REVIEW

File: spec/tracebacks_spec.lua [hunk 1]
```diff
@@ -3,14 +3,15 @@
 -- Please refer to the LICENSE and AUTHORS files for details
 -- SPDX-License-Identifier: MIT
 
-local util = require "misc.util"
+local util = require "spec.util"
```
Comment (singul4ri7y) [inline spec/tracebacks_spec.lua:6]: Now that `util.lua` is out of the way, can we expect the `pt-run` source code to be in `src` directory as a single file, e.g. `pt-run.c` or `main.c`?

File: spec/tracebacks_spec.lua [hunk 2]
```diff
@@ -3,14 +3,15 @@
...
 local function assert_test(example, expected_content)
-    local cdir = "spec/tracebacks/"..example.."/";
-    local ok, err = util.execute("cd "..cdir.."&& make")
+    local cdir = util.shell_quote("spec/tracebacks/"..example)
+    local ok, err = util.execute(string.format("cd %s && make", cdir))
     assert(ok, err)
 
-    local ok, _, output_content, err_content = util.outputs_of_execute("cd "..cdir.." && pt-run main.lua")
+    local ok, _, output_content, err_content =
+        util.outputs_of_execute(string.format("cd %s && pt-run main.lua", cdir))
```
Comment (singul4ri7y) [inline spec/tracebacks_spec.lua:14]: Any particular reason to explicitly call to `string.format`? We are not doing any complex string formatting, shouldn't concatenation suffice and cleaner in this case?

File: spec/tracebacks_spec.lua (hunk 1 above)
Comment (hugomg) [reply spec/tracebacks_spec.lua:6]: I'm working on a PR to fix the makefile and will decide based on what makes things easier for that. Currently, src/pt-run/main.c --> src/pt-run.c  and lib/ptracer.h --> src/ptracer.h. But that's not set in stone yet.

File: spec/tracebacks_spec.lua (hunk 2 above)
Comment (hugomg) [reply spec/tracebacks_spec.lua:14]: I think with string.format it's easier to see the spaces between the words. In the previous version we were forgetting to add spaces before the "&&"!

    "cd "..cdir.."&&"

**Review by singul4ri7y — APPROVED** [review]

**Pull request merged**